src/core/lib/iomgr/error.h

```diff
@@ -136,13 +136,15 @@ typedef enum {
   GRPC_ERROR_TIME_CREATED,
 } grpc_error_times;
 
+
 /// The following "special" errors can be propagated without allocating memory.
 /// They are always even so that other code (particularly combiner locks) can
 /// safely use the lower bit for themselves.
 
 #define GRPC_ERROR_NONE ((grpc_error *)NULL)
 #define GRPC_ERROR_OOM ((grpc_error *)2)
 #define GRPC_ERROR_CANCELLED ((grpc_error *)4)
+#define GRPC_ERROR_INTERNAL ((grpc_error *)8)
```

Comment:
Yes - but then we ideally should not have any 'special' error like `(grpc_error *) 1` in future; because we would not be able to distinguish between a legitimate error versus CLOSURE_SHUTDOWN_BIT if `fd_shutdown` is called with `why == (grpc_error *) 1`.  This is not a big deal (its just that we miss reporting the correct error for debugging purposes) but I felt that adding a special error was the correct way to do this.

I can rename `GRPC_ERROR_INTERNAL` to `GRPC_FD_SHUTDOWN` and make it `(grpc_error *) 1` - if you like that (that was my first choice but something stopped me :) )